cmd/kueuectl/app/util/completion.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"strings"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	"sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+func NamespaceNameCompletionFunc(clientGetter ClientGetter) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		clientSet, err := clientGetter.K8sClientSet()
+		if err != nil {
+			return []string{}, cobra.ShellCompDirectiveError
+		}
+
+		list, err := clientSet.CoreV1().Namespaces().List(cmd.Context(), metav1.ListOptions{})
+		if err != nil {
+			return []string{}, cobra.ShellCompDirectiveError
+		}
+
+		validArgs := make([]string, len(list.Items))
+		for i, wl := range list.Items {
+			validArgs[i] = wl.Name
```

Comment:
Don't you also need to add `if strings.HasPrefix(name, toComplete)`?

Reply:
No, it's not necessary. Cobra filtering it.